tsfile/src/main/java/org/apache/iotdb/tsfile/compress/IUnCompressor.java

```diff
@@ -184,4 +186,82 @@ public CompressionType getCodecName() {
       return CompressionType.SNAPPY;
     }
   }
+
+
+  class LZ4UnCompressor implements IUnCompressor {
+
+    private static final Logger logger = LoggerFactory.getLogger(LZ4Compressor.class);
+    private static final int MAX_COMPRESS_RATIO = 255;
+    private LZ4SafeDecompressor decompressor;
+
+
+    public LZ4UnCompressor() {
+      LZ4Factory factory = LZ4Factory.fastestInstance();
+      decompressor = factory.safeDecompressor();
+    }
+
+    @Override
+    public int getUncompressedLength(byte[] array, int offset, int length) throws IOException {
+      throw new UnsupportedOperationException("unsupported get uncompress length");
+    }
+
+    @Override
+    public int getUncompressedLength(ByteBuffer buffer) throws IOException {
+      throw new UnsupportedOperationException("unsupported get uncompress length");
+    }
+
+    /**
+     * We don't recommend using this method because we have to allocate MAX_COMPRESS_RATIO *
+     * compressedSize to ensure uncompress safety, you can use other method if you know the
+     * uncompressed size
+     */
+    @Override
+    public byte[] uncompress(byte[] bytes) {
+      if (bytes == null) {
+        return new byte[0];
+      }
+
+      try {
+        return decompressor.decompress(bytes, MAX_COMPRESS_RATIO * bytes.length);
+      } catch (RuntimeException e) {
+        logger.error(
```

Comment:
I think it's better to throw this exception and let the upper level handle it.

Reply:
Sure~